tensorflow/core/transforms/remapper/remapping_helper.h

```diff
@@ -0,0 +1,260 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_CORE_TRANSFORMS_REMAPPER_REMAPPING_HELPER_H_
+#define TENSORFLOW_CORE_TRANSFORMS_REMAPPER_REMAPPING_HELPER_H_
+
+#include "tensorflow/core/transforms/utils/op_cat_helper.h"
+#include "tensorflow/core/transforms/utils/utils.h"
+
+namespace mlir {
+namespace tfg {
+namespace remapping {
+
+struct ContractionBiasAdd {
+  Operation* contraction;
+  Operation* bias_add;
+};
+
+struct ContractionBiasAddActivation {
+  Operation* contraction;
+  Operation* bias_add;
+  Operation* activation;
+};
+
+struct ContractionBiasAddAdd {
+  Operation* contraction;
+  Operation* bias_add;
+  Operation* add;
+};
+
+struct ContractionBiasAddAddActivation {
+  Operation* contraction;
+  Operation* bias_add;
+  Operation* add;
+  Operation* activation;
+};
+
+// This enum class is used as a template parameter and meant for alias to tfg op
+// name.
+// TODO(intel-tf): Add more items as needed.
+enum class OpKind { Relu, Relu6, Elu, LeakyRelu, Tanh, Sigmoid };
+
+inline std::string GetTfgOpName(OpKind op_kind) {
+  switch (op_kind) {
+    case OpKind::Relu:
+      return "tfg.Relu";
+    case OpKind::Relu6:
+      return "tfg.Relu6";
+    case OpKind::Elu:
+      return "tfg.Elu";
+    case OpKind::LeakyRelu:
+      return "tfg.LeakyRelu";
+    case OpKind::Tanh:
+      return "tfg.Tanh";
+    case OpKind::Sigmoid:
+      return "tfg.Sigmoid";
+    default:
+      return "tfg.NoOp";
+  }
+}
+
+class OpPropertyHelper : public OpCatHelper {
+ public:
+  OpPropertyHelper(TFGraphDialect* dialect, bool onednn_enabled = false,
+                   bool xla_auto_clustering = false)
+      : OpCatHelper(dialect),
+        dialect_(dialect),
+        is_onednn_enabled_(onednn_enabled),
+        xla_auto_clustering_(xla_auto_clustering) {}
+
+  // Get the TFG dialect instance.
+  TFGraphDialect* getDialect() { return dialect_; }
```

Comment:
Seems like it would be suitable on the base class?

Reply:
Moved to the base class.